modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsConsumer.java

```diff
@@ -63,67 +74,56 @@ public JmsConsumer(JmsStore store) {
     }
 
     public MessageContext receive() {
-        boolean error;
-        JMSException exception;
-        if (!checkConnection()) {
-            if (!reconnect()) {
-                if (logger.isDebugEnabled()) {
-                    logger.debug(getId()
-                            + " cannot receive message from store. Cannot reconnect.");
-                }
-                return null;
-            } else {
-                logger.info(getId() + " reconnected to store.");
-                isReceiveError = false;
-            }
 
+        boolean connectionSuccess = checkAndTryConnect();
+
+        if(!connectionSuccess) {
+            throw new SynapseException(idString + "Error while connecting to JMS provider. " +
+                    MessageProcessorConstants.STORE_CONNECTION_ERROR);
```

Comment:
Operator should precede the line shall we fix this?

Reply:
Fixed.